pkg/security/admission/admission.go

```diff
@@ -183,24 +183,29 @@ func assignSecurityContext(provider scc.SecurityContextConstraintsProvider, pod
 
 	// Note: this is not changing the original container, we will set container SCs later so long
 	// as all containers validated under the same SCC.
-	for i, containerCopy := range pod.Spec.Containers {
-		// We will determine the effective security context for the container and validate against that
-		// since that is how the sc provider will eventually apply settings in the runtime.
-		// This results in an SC that is based on the Pod's PSC with the set fields from the container
-		// overriding pod level settings.
-		containerCopy.SecurityContext = sc.DetermineEffectiveSecurityContext(pod, &containerCopy)
-
-		sc, err := provider.CreateContainerSecurityContext(pod, &containerCopy)
-		if err != nil {
-			errs = append(errs, field.Invalid(field.NewPath("spec", "containers").Index(i).Child("securityContext"), "", err.Error()))
+	containerPath := field.NewPath("spec", "initContainers")
+	for i, containerCopy := range pod.Spec.InitContainers {
+		csc, resolutionErrs := resolveContainerSecurityContext(provider, pod, &containerCopy, containerPath.Index(i))
+		errs = append(errs, resolutionErrs...)
+		if csc == nil {
 			continue
 		}
-		generatedSCs[i] = sc
-
-		containerCopy.SecurityContext = sc
-		errs = append(errs, provider.ValidateContainerSecurityContext(pod, &containerCopy, field.NewPath("spec", "containers").Index(i).Child("securityContext"))...)
+		generatedSCs[i] = csc
 	}
 
+	base := len(pod.Spec.InitContainers)
+
+	// Note: this is not changing the original container, we will set container SCs later so long
+	// as all containers validated under the same SCC.
+	containerPath = field.NewPath("spec", "containers")
+	for i, containerCopy := range pod.Spec.Containers {
+		csc, resolutionErrs := resolveContainerSecurityContext(provider, pod, &containerCopy, containerPath.Index(i))
+		errs = append(errs, resolutionErrs...)
+		if csc == nil {
```

Comment:
seems safer to check errs here instead rather than setting a `csc` that didn't happen to be `nil` but also didn't validate.  Shouldn't happen but technically could.